modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/shibari_stand.dm

```diff
@@ -0,0 +1,269 @@
+			L.set_pull_offsets(buckledmeme, L.grab_state)
+	unbuckle_mob(buckled_mob)
+	return buckledmeme
+
+/obj/structure/chair/shibari_stand/user_buckle_mob(mob/living/buckledmeme, mob/user, check_loc = TRUE)
+	if(!is_user_buckle_possible(buckledmeme, user, check_loc))
+		return FALSE
+	add_fingerprint(user)
+
+	if(!ishuman(buckledmeme))
+		return FALSE
+
+	var/mob/living/carbon/human/hooman = buckledmeme
+	if(!(istype(hooman.w_uniform, /obj/item/clothing/under/shibari_fullbody)))
+		to_chat(user, span_warning("You'll need to completely tie their body!"))
+		return FALSE
+	if(!istype(user.get_active_held_item(), /obj/item/stack/shibari_rope))
+		to_chat(user, span_warning("You'll need to be holding shibari ropes to tie them to the stand!!"))
+		return FALSE
+
+	if(buckledmeme != user)
+		buckledmeme.visible_message(span_warning("[user] starts tying [buckledmeme] to [src]!"),\
+			span_userdanger("[user] starts tying you to [src]!"),\
+			span_hear("You hear ropes being tightened."))
+		if((HAS_TRAIT(user, TRAIT_RIGGER)))
+			if(!do_after(user, 5 SECONDS, buckledmeme))
+				return FALSE
+		else
+			if(!do_after(user, 10 SECONDS, buckledmeme))
+				return FALSE
+
+		if(!is_user_buckle_possible(buckledmeme, user, check_loc))
+			return FALSE
+
+		if(!(istype(hooman.w_uniform, /obj/item/clothing/under/shibari_fullbody)))
+			to_chat(user, span_warning("You'll need to completely tie their body!"))
+			return FALSE
+		if(!istype(user.get_active_held_item(), /obj/item/stack/shibari_rope))
+			to_chat(user, span_warning("You'll need to be holding shibari ropes to tie them to the stand!"))
+			return FALSE
+
+		if(buckle_mob(buckledmeme, check_loc = check_loc))
+			var/obj/item/stack/shibari_rope/rope = user.get_active_held_item()
+			ropee = new()
+			ropee.current_color = rope.current_color
+			ropee.update_icon_state()
+			ropee.update_icon()
+			rope.use(1)
+			add_overlay(shibari_shadow_overlay)
+			add_rope_overlays(ropee.current_color, hooman?.dna?.species?.mutant_bodyparts["taur"])
+			buckledmeme.visible_message(span_warning("[user] tied [buckledmeme] to [src]!"),\
+				span_userdanger("[user] tied you to [src]!"),\
+				span_hear("You hear ropes being completely tightened."))
+	else
+		to_chat(user, span_warning("You cannot buckle yourself to this stand, there is no way that level of self-bondage exists!"))
+		return FALSE
+
+/obj/structure/chair/shibari_stand/deconstruct()
+	qdel(src)
+	return TRUE
+
+/obj/structure/chair/shibari_stand/proc/add_rope_overlays(color, taur)
+	cut_overlay(shibari_rope_overlay)
+	cut_overlay(shibari_rope_overlay_behind)
+	if(taur)
+		shibari_rope_overlay = mutable_appearance('modular_skyrat/modules/modular_items/lewd_items/icons/obj/lewd_structures/shibari_stand.dmi', "ropes_above_[color]_snek", ABOVE_MOB_LAYER)
+	else
+		shibari_rope_overlay = mutable_appearance('modular_skyrat/modules/modular_items/lewd_items/icons/obj/lewd_structures/shibari_stand.dmi', "ropes_above_[color]", ABOVE_MOB_LAYER)
```

Comment:
Just use a ternary operator for the icon_state instead of that much copy-paste.

Also, consider making some defines for those .dmi file paths because they're incredibly long and repeated everywhere. Define them at the start of the file and then do `#undef DEFINE_NAME_HERE` at the end of the file so it doesn't pollute the global codespace.